### ISSUE TITLE
2.6.5 release breaks JDK7 compatibility

### ISSUE THREAD

**Issue by mulderbaba** [eclipse-ee4j/eclipselink#164]
commit a49fe7bb34b881fdb86dbd10cd7d9431844c2117 upgrades asm_5.0.1.v201405080102 to asm_6.0.0.v201702131300. This commit also resides within 2.6.5 release which breaks JDK7 compatibility since asm_6.0.0.v201702131300 requires JDK8 at least

**Comment by mulderbaba** [comment]
@aguibert Actually, It's not the ASM but the org.eclipse.persistence.asm artifact that eclipselink creates. I found the core reason as: plugins/org.eclipse.persistence.asm is not building right on 2.6.5 tag release and spits out compilation errors (due to missing commit 1775741f577b9af6f55b3d425e08e7d4842fc615 Bug 493817 - Upgrade to ASM 5.1.0). I might accidentally have used 2.7.x built of org.eclipse.persistence.asm, double checking it now. and yes that would require JDK1-8 because of the MANIFEST entries.

**Comment by mulderbaba** [comment]
closing this one since the cause of the problem is org.eclipse.persistence.asm is not building right on 2.6.5 tag since 2.6.x branches miss the commit:
```
mertcaliskan:eclipselink mertcaliskan$ git tag --contains 1775741f577b9af6f55b3d425e08e7d4842fc615
2.7.0
2.7.0-RC1
2.7.0-RC2
2.7.0-RC3
2.7.1
2.7.1-RC1
2.7.2
2.7.2-RC1
2.7.2-RC2
```

**Comment by mulderbaba** [comment]
reopening since released artifacts contains wrong directives in MANIFEST file. Artifact w/ GAV:
```
<dependency>
    <groupId>org.eclipse.persistence</groupId>
    <artifactId>org.eclipse.persistence.asm</artifactId>
    <version>2.6.5</version>
</dependency>
```
contains
```
Bundle-RequiredExecutionEnvironment: JavaSE-1.8
```

**Comment by lukasj** [comment]
closed by #204
